### PR TITLE
Feat: add auto approve field to sub environment configuration

### LINKED ISSUE #869: Feat: Add auto approve field to sub environment configuration
**Is your feature request related to a problem? Please describe.**
We now support workflow approval flows in the API, we should also have support to it from the provider

**Describe the solution you'd like**
We want to add a `approve_plan_automatically` to the [sub_environment_configuration](https://registry.terraform.io/providers/env0/env0/1.5.0/docs/resources/environment#nested-schema-for-sub_environment_configuration) object so users will be able to use workflow approval flows from the provider

the API is the same for create&deploy the added field can be seen in [the api](https://docs.env0.com/reference/environments-deploy) under the subEnvironments object as `userRequiresApproval` for the sub environment object

### PR DESCRIPTION
**Pull request by TomerHeber**
### Issue & Steps to Reproduce / Feature Request
resolves #869 

### Solution

1. Added the new field to the API.
2. Added the new field to the schema.
3. Updated the code.
4. Updated the acceptance tests.
5. Updated the integration tests.

### CODE REVIEW

File: env0/resource_environment.go [hunk 1]
```diff
@@ -744,6 +751,7 @@ func deploy(d *schema.ResourceData, apiClient client.ApiClientInterface) diag.Di
 				Revision:             subEnvironment.Revision,
 				Workspace:            subEnvironment.Workspace,
 				ConfigurationChanges: configurationChanges,
+				UserRequiresApproval: !subEnvironment.ApprovePlanAutomatically,
```
Comment (TomerHeber) [inline env0/resource_environment.go:754]: notice the '!'

File: env0/resource_environment.go [hunk 2]
```diff
@@ -885,6 +893,7 @@ func getCreatePayload(d *schema.ResourceData, apiClient client.ApiClientInterfac
 				Revision:             subEnvironment.Revision,
 				ConfigurationChanges: subEnvironment.Configuration,
 				Workspace:            subEnvironment.Workspace,
+				UserRequiresApproval: !subEnvironment.ApprovePlanAutomatically,
```
Comment (TomerHeber) [inline env0/resource_environment.go:896]: notice the '!'

File: env0/resource_environment_test.go [hunk 3]
```diff
@@ -2437,6 +2438,7 @@ func TestUnitEnvironmentWithSubEnvironment(t *testing.T) {
 				Revision:             subEnvironment.Revision,
 				Workspace:            updatedSubEnvironment.Workspace,
 				ConfigurationChanges: updatedSubEnvironment.Configuration,
+				UserRequiresApproval: true,
```
Comment (TomerHeber) [inline env0/resource_environment_test.go:2441]: `false` for 'approve_plan_automatically' will set `UserRequiresApproval` to `true`

File: env0/resource_environment.go (hunk 1 above)
Comment (ItamarMalka) [reply env0/resource_environment.go:754]: so what will happen when it is not there?

**Comment by ItamarMalka** [comment]
[Open ticket to add approval for sub envs in the provider](https://trello.com/c/5rCxvFmi/13270-open-ticket-to-add-approval-for-sub-envs-in-the-provider)

File: env0/resource_environment.go (hunk 1 above)
Comment (TomerHeber) [reply env0/resource_environment.go:754]: the schema has a default value 'true'.
so it will be 'false'.

**Review by avnerenv0 — APPROVED** [review]
LGTM, 
Please try to fix all the typos in a different PR cause most of the changes here are not related.

**Comment by TomerHeber** [comment]
> LGTM, Please try to fix all the typos in a different PR cause most of the changes here are not related.

My aplogies.
I will refrain from doing that moving forward.
I'll merge this PR (to avoid the headache of reverting all the typo changes).

**Pull request merged**